module-4/metascan/metascan.py

```diff
@@ -0,0 +1,91 @@
+import os
+import csv
+import requests
+from bs4 import BeautifulSoup
+from urllib.parse import urljoin, urlparse
+from PyPDF2 import PdfReader
+import argparse
+
+def extract_pdf_links(url):
+    response = requests.get(url)
+    response.raise_for_status()
+    soup = BeautifulSoup(response.text, 'html.parser')
+    pdf_links = [urljoin(url, link.get('href')) for link in soup.find_all('a', href=True) if link.get('href').endswith('.pdf')]
```

Comment:
better do that in two separate lines for readability reasons and improved debugging opportunities.

e.g.
```python
html_links = soup.find_all('a', href=True)
pdf_links = filter(
    lambda x: x.get("href").endswith(".pdf"),
    html_links)
pdf_urls = list(
    map(lambda x: urljoin(url, x), pdf_links))
```